### ISSUE TITLE
pip install results in a lot of errors

### ISSUE THREAD

**Issue by samfux84** [rusty1s/pytorch_scatter#10]
Hi,

I am trying to install torch-scatter on our HPC cluster for Python 3.6.4 using GCC 6.3.0.

When running

`pip install --user torch-scatter`

or

`pip install --user --no-binary :all: torch-scatter`

Then it fails with a lot of errors like for instance

```
    cpu/dim_apply.h:20:55: error: invalid initialization of reference of type ‘const at::Type&’ from expression of type ‘long int’
         auto zeros = at::zeros(dims, torch::CPU(at::kLong));
```

I have attached the full log of the "pip install" command in the file pip_install.log.

[pip_install.log](https://github.com/rusty1s/pytorch_scatter/files/2537844/pip_install.log)

Is this a known problem ?

It seems to be similar to the following problem discussed on stackoverflow:

https://stackoverflow.com/questions/11371216/error-invalid-initialization-of-reference-of-type-int-from-expression-of-typ

Best regards

Sam

**Comment by samfux84** [comment]
Some additional information:

OS:

```
[sfux@lo-s4-028 ~]$ lsb_release -a
LSB Version:    :core-4.1-amd64:core-4.1-noarch
Distributor ID: CentOS
Description:    CentOS Linux release 7.5.1804 (Core) 
Release:        7.5.1804
Codename:       Core
```

Modules loaded:

```
[sfux@lo-s4-028 ~]$ module list
Currently Loaded Modules:
  1) StdEnv      3) gcc/6.3.0         5) cuda/9.0.176   7) jpeg/9b         9) python_gpu/3.6.4
  2) eth_proxy   4) openblas/0.2.19   6) cudnn/7.0      8) libpng/1.6.27
```

CPATH and PATH:

[sfux@lo-s4-028 ~]$ **echo $PATH**
/cluster/apps/gcc-6.3.0/libpng-1.6.27-7vevbsio2diw5xquq7oxg7fphkppf2kx/bin:/cluster/apps/gcc-6.3.0/jpeg-9b-742nyforj7mfuhcfzbsbqjlmiya2wi5e/bin:**/cluster/apps/gcc-6.3.0/cuda-9.0.176-fpkau3eaasshumsn6it7cn2cbdzonydn/bin**:/cluster/apps/gcc-6.3.0/openblas-0.2.19-jusdys62qhv52qjbsoewvthriekuaynw/bin:/cluster/apps/python/3.6.4/bin:/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/gcc-6.3.0-d7fey5ouwb3cw26w5zqafqcse7q3fs6d/bin:/cluster/home/sfux/spacktest/spack/bin:/cluster/apps/lsf/10.1/linux3.10-glibc2.17-x86_64/bin:/cluster/apps/lsf/10.1/linux3.10-glibc2.17-x86_64/bin:/cluster/home/sfux/spacktest/spack/bin:/cluster/apps/sfos/bin:/usr/local/bin:/usr/bin:/usr/local/sbin:/usr/sbin:/opt/ibutils/bin:/cluster/home/sfux/.local/bin:/cluster/home/sfux/bin:/cluster/apps/adm:/cluster/home/sfux/bin
[sfux@lo-s4-028 ~]$ **echo $CPATH**
/cluster/apps/gcc-6.3.0/libpng-1.6.27-7vevbsio2diw5xquq7oxg7fphkppf2kx/include:/cluster/apps/gcc-6.3.0/jpeg-9b-742nyforj7mfuhcfzbsbqjlmiya2wi5e/include:/cluster/apps/gcc-6.3.0/cudnn-7.0-dbcgz3tfu4fqn2pukryudzbd2l55f6dc/include:**/cluster/apps/gcc-6.3.0/cuda-9.0.176-fpkau3eaasshumsn6it7cn2cbdzonydn/include**:/cluster/apps/gcc-6.3.0/openblas-0.2.19-jusdys62qhv52qjbsoewvthriekuaynw/include:/cluster/apps/python/3.6.4/include:/cluster/spack/apps/linux-centos7-x86_64/gcc-4.8.5/gcc-6.3.0-d7fey5ouwb3cw26w5zqafqcse7q3fs6d/include

GPU:

```
[sfux@lo-s4-028 ~]$ nvidia-smi
Thu Nov  1 10:10:35 2018       
+-----------------------------------------------------------------------------+
| NVIDIA-SMI 396.26                 Driver Version: 396.26                    |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|===============================+======================+======================|
|   0  GeForce GTX 108...  Off  | 00000000:07:00.0 Off |                  N/A |
| 20%   25C    P8    15W / 250W |      0MiB / 11178MiB |      0%   E. Process |
+-------------------------------+----------------------+----------------------+
                                                                               
+-----------------------------------------------------------------------------+
| Processes:                                                       GPU Memory |
|  GPU       PID   Type   Process name                             Usage      |
|=============================================================================|
|  No running processes found                                                 |
+-----------------------------------------------------------------------------+
```

**Comment by rusty1s** [comment]
Which PyTorch version are you using?

**Comment by samfux84** [comment]
commit 75dbf9b113a8683a22597f8c78e572f5e4cbb864 from the github repository

**Comment by rusty1s** [comment]
I mean, which PyTorch version? 0.4.0? 0.4.1? 1.0.0?

**Comment by samfux84** [comment]
Hard to say which version commit 75dbf9b113a8683a22597f8c78e572f5e4cbb864 corresponds to.

troch.__version__  is 0.5.0a0, but this is not a proper release, therefore I am providing the git commit string.

```
[sfux@lo-login-02 ~]$ python
Python 3.6.4 (default, Apr 10 2018, 08:00:27) 
[GCC 4.8.5 20150623 (Red Hat 4.8.5-16)] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.__version__
'0.5.0a0+75dbf9b'
>>> 
```

**Comment by rusty1s** [comment]
Sorry for no reply for a long time. I totally missed this issue. Does this got resolved (e.g. by updating to PyTorch 1.0)?

**Comment by samfux84** [comment]
Yes

**Comment by rusty1s** [comment]
What did you do to resolve it?

**Comment by samfux84** [comment]
It was resolved by using PyTorch 1.0
